packages/@lwc/engine/src/framework/restrictions.ts

```diff
@@ -447,6 +490,6 @@ export function patchComponentWithRestrictions(cmp: ComponentInterface) {
     defineProperties(cmp, getComponentRestrictionsDescriptors());
 }
 
-export function patchLightningElementPrototypeWithRestrictions(proto: object) {
+export function patchLightningElementPrototypeWithRestrictions(proto: LightningElement) {
```

Comment:
I don't think `proto` is a `LightningElement`, it is the prototype of the LightningElement constructor, which is different.

Reply:
I did update the type to `typeof LightningElement.prototype` that being said, it's theoretically the same as `LightningElement`: [type definition](https://github.com/salesforce/lwc/blob/master/packages/@lwc/engine/src/framework/base-lightning-element.ts#L148). 

Let's keep it like this for now and improve the typings over time, there are bigger fishes to fry in terms of type correctness in the engine for now (typescript strict mode is still partially enabled in the engine).